### PR TITLE
Symfony 5 support & refactoring

### PR DESCRIPTION
**Pull request by MichaelKubovic**
- added support for symfony 5
- added option to skip HttpListener for configured routes (e.g. health_check)
- refactored service factory
- added messenger listener

### CODE REVIEW

**Review by dejwCake — APPROVED** [review]

File: src/EventListener/MessengerListener.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        $this->tracer = $tracer;
+        $this->nameGetter = $nameGetter;
+    }
+
+    public function onWorkerMessageReceived(WorkerMessageReceivedEvent $event)
+    {
+        $envelope = $event->getEnvelope();
+        $name = $this->nameGetter->getNameByMessage($envelope);
+        $this->tracer->startActiveSpan($name);
+        $this->tracer->getActiveSpan()->setTag('messenger.message', get_class($envelope->getMessage()));
```
Comment (MatokUK) [inline src/EventListener/MessengerListener.php]:
```suggestion
        $this->tracer->startActiveSpan($name, [
            'tags' => ['messenger.message' => \get_class($envelope->getMessage()],
        ]);
```

I'm not 100% sure but this should do the same job.

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -45,6 +45,9 @@ public function getConfigTreeBuilder()
                         ->scalarNode('debug_id_header_key')->defaultNull()->end()
                     ->end()
                 ->end()
+                ->arrayNode('http_listener_skipped_routes')
+                    ->prototype('scalar')->end()
+                ->end()
```
Comment (MatokUK) [inline src/DependencyInjection/Configuration.php]:
```suggestion
                ->arrayNode('http_listener_skipped_routes')
                    ->beforeNormalization()
                        ->castToArray()
                    ->end()
                    ->prototype('scalar')->end()
                ->end()
```

after this you can write config like this:

```yml
  http_listener_skipped_routes: health
```

or like this
```yml
  http_listener_skipped_routes: 
    - health
    - some_other_route
```

And is goona be an array in both cases

File: src/DependencyInjection/SymfonyOpenTracingExtension.php [hunk 3]
```diff
@@ -19,8 +19,23 @@ protected function loadInternal(array $mergedConfig, ContainerBuilder $container
...
-        $container->setParameter('symfony_open_tracing.service_name', $mergedConfig['service_name']);
-        $container->setParameter('symfony_open_tracing.enabled', $mergedConfig['enabled']);
-        $container->setParameter('symfony_open_tracing.tracer_config', $mergedConfig['tracer_config']);
+        if ($mergedConfig['enabled']) {
+            $this->configureJaegerTracer($container, $mergedConfig);
+        }
+
+        $container
+            ->getDefinition('open_tracing.http_listener')
+            ->setArgument('$skippedRoutes', $mergedConfig['http_listener_skipped_routes'] ?? []);
```
Comment (MatokUK) [inline src/DependencyInjection/SymfonyOpenTracingExtension.php]: `$mergedConfig['http_listener_skipped_routes']` will be `[]` (at least after suggestion above, that this fallback is not necassary)

File: src/Service/RootSpanNameBuilder.php [hunk 4]
```diff
@@ -24,15 +27,18 @@ class RootSpanNameBuilder implements GetSpanNameByRequest, GetSpanNameByCommand
...
-     *
-     * @param string $httpNamePrefix
-     * @param string $cliNamePrefix
+     * @var string
      */
-    public function __construct(string $httpNamePrefix = 'http-tracing', string $cliNamePrefix = 'cli-tracing')
-    {
+    private $messageNamePrefix;
+
+    public function __construct(
```
Comment (MatokUK) [inline src/Service/RootSpanNameBuilder.php]: fyi: this is violation of SF CS:

> Declare all the arguments on the same line as the method/function name, no matter how many arguments there are;

File: src/EventListener/HttpListener.php [hunk 5]
```diff
@@ -5,119 +5,139 @@
...
             $span->finish();
         }
 
-        $this->openTracing->getTracer()->flush();
+        $this->tracer->flush();
+    }
+
+    private function skipRequest(Request $request): bool
+    {
+        return in_array($request->attributes->get('_route'), $this->skippedRoutes);
```
Comment (MatokUK) [inline src/EventListener/HttpListener.php]: It's not necessary but for better performance (according to some benchmarks) and/or "code cleanness/**purpose**" you should consider using `isset` by application:

1. create associative array in the Extension (eg `array_combine($mergedConfig['http_listener_skipped_routes'], $mergedConfig['http_listener_skipped_routes'])`)
2. set this accos. array to `'open_tracing.http_listener'` (1&2 == "compile time")
3. use `return isset($this->skippedRoutes[$request->attributes->get('_route')]);`

Main point of this is removing _possible duplications_ in route configuration list ...or ... Did you consider this kind of validation in the Extension?

Comment (MichaelKubovic) [reply src/EventListener/HttpListener.php]: valid point, I've used array_flip in the extension

File: src/Service/RootSpanNameBuilder.php (hunk 4 above)
Comment (MichaelKubovic) [reply src/Service/RootSpanNameBuilder.php]: fixed

File: src/DependencyInjection/SymfonyOpenTracingExtension.php (hunk 3 above)
Comment (MichaelKubovic) [reply src/DependencyInjection/SymfonyOpenTracingExtension.php]: right, fixed!

File: src/DependencyInjection/Configuration.php (hunk 2 above)
Comment (MichaelKubovic) [reply src/DependencyInjection/Configuration.php]: done

File: src/EventListener/MessengerListener.php (hunk 1 above)
Comment (MichaelKubovic) [reply src/EventListener/MessengerListener.php]: yeap, I've combined both method calls

**Pull request merged**